spec/unit/view_spec.rb

```diff
@@ -64,8 +64,30 @@ def title
       }.new
     }
 
-    it "raises error when layout cannot be found" do
-      expect { view.() }.to raise_error Hanami::View::LayoutNotFoundError, /Layout \+missing_layout\+/
+    it "raises a LayoutNotFoundError error when layout cannot be found" do
+      expect { view.() }.to raise_error Hanami::View::TemplateNotFoundError, %r{Template `layouts/missing_layout' for format `html' could not be found}
+    end
+  end
+
+  describe "template rendering" do
+    it "raises a TemplateNotFoundError when the template cannot be found" do
+      view = Class.new(Hanami::View) {
+        config.paths = SPEC_ROOT.join("fixtures/templates")
+        config.layout = nil
+        config.template = "missing_template"
+      }.new
+
+      expect { view.() }.to raise_error Hanami::View::TemplateNotFoundError, /Template `missing_template' for format `html' could not be found/
```

Comment:
@timriley In case of Hanami app, will the real error message appear with the relative path of the template?

Example:

```
Template `app/templates/layouts/foo.html.erb' for format `html' could not be found
```

As a dev, I can just copy and paste the path and create the missing template.

Reply:
@jodosha Here's the exact error:

```
     Hanami::View::TemplateNotFoundError:
       Template `layouts/app' for format `html' could not be found in paths:

        - /Users/tim/Source/scratch/bookshelf/app/templates
```

So yes, you could join `layouts/app` onto `/Users/tim/Source/scratch/bookshelf/app/templates` and there you have the correct path to use (minus the file extension).

We can't put an _exact_ file extension on this because we do support fuzzy loading e.g. from any `*.html.*` template files.

I think we _could_ probably finesse this error message to make it even nicer. For example, if there's only one path configured, then we could make the error a lot closer to your example.

However, I'd like to leave that as an exercise for a future enhancement rather than block on this now. This would be a great first issue for a potential Hanami contributor, for example 😄